android/src/main/java/com/swmansion/rnscreens/RNScreensPackage.kt

```diff
@@ -51,4 +104,120 @@ class RNScreensPackage : TurboReactPackage() {
             moduleInfos
         }
     }
+
+    /**
+     * Initializes dummy view hierarchy with CoordinatorLayout, AppBarLayout and dummy View.
+     * We utilize this to compute header height (app bar layout height) from C++ layer when its needed.
+     */
+    private fun ensureDummyLayoutWithHeader(reactContext: ReactApplicationContext) {
+        if (::coordinatorLayout.isInitialized) {
+            return
+        }
+
+        // We need to use activity here, as react context does not have theme attributes required by
+        // AppBarLayout attached leading to crash.
+        val contextWithTheme =
+            requireNotNull(reactContext.currentActivity) { "[RNScreens] Attempt to use context detached from activity" }
+
+        coordinatorLayout = CoordinatorLayout(contextWithTheme)
+
+        appBarLayout = AppBarLayout(contextWithTheme).apply {
+            layoutParams = CoordinatorLayout.LayoutParams(
+                CoordinatorLayout.LayoutParams.MATCH_PARENT,
+                CoordinatorLayout.LayoutParams.WRAP_CONTENT,
+            )
+        }
+
+        toolbar = Toolbar(contextWithTheme).apply {
+            title = "FontSize123!#$"
+            layoutParams = AppBarLayout.LayoutParams(
+                AppBarLayout.LayoutParams.MATCH_PARENT,
+                AppBarLayout.LayoutParams.WRAP_CONTENT
+            ).apply { scrollFlags = 0 }
+        }
+
+        // We know the title text view will be there, cause we've just set title.
+        defaultFontSize = ScreenStackHeaderConfig.findTitleTextViewInToolbar(toolbar)!!.textSize
+
+        appBarLayout.addView(toolbar)
+
+        dummyContentView = View(contextWithTheme).apply {
+            layoutParams = CoordinatorLayout.LayoutParams(
+                CoordinatorLayout.LayoutParams.MATCH_PARENT,
+                CoordinatorLayout.LayoutParams.MATCH_PARENT
+            )
+        }
+
+        coordinatorLayout.apply {
+            addView(appBarLayout)
+            addView(dummyContentView)
+        }
+    }
+
+    /**
+     * Triggers layout pass on dummy view hierarchy, taking into consideration selected
+     * ScreenStackHeaderConfig props that might have impact on final header height.
+     *
+     * @param fontSize font size value as passed from JS
+     * @return header height in dp as consumed by Yoga
+     */
+    private fun computeDummyLayout(fontSize: Int): Float {
```

Comment:
I think we could move the implementation of those methods to some utility file and only leave what's 100% needed here to keep the code cleaner.

Reply:
https://github.com/software-mansion/react-native-screens/pull/2169/commits/d3da35727ffaba855814cd6a8556087961c1a128